test/integration/apiserver/watchcache_test.go

```diff
@@ -234,3 +238,64 @@ func BenchmarkListFromWatchCache(b *testing.B) {
 		b.Logf("Number of secrets: %d", len(secrets.Items))
 	}
 }
+
+func TestConsistentList(t *testing.T) {
+	var testcases = []struct {
+		name      string
+		fromCache bool
+	}{
+		{
+			name:      "FromCache",
+			fromCache: true,
+		},
+		{
+			name:      "FromStorage",
+			fromCache: false,
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			_, ctx := ktesting.NewTestContext(t)
+			ctx, cancel := context.WithCancel(ctx)
+			defer cancel()
+
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.ConsistentListFromCache, tc.fromCache)()
+
+			clientSet, _, tearDownFn := framework.StartTestServer(ctx, t, framework.TestServerSetup{})
```

Comment:
Don't we need to increase bookmark interval?

We may have some races where it's actually delivered + more importantly we rely in this test on interval in https://github.com/kubernetes/kubernetes/blob/b9887bec3da5d5ec5fbb5320291410c7f0c9f558/test/integration/apiserver/watchcache_test.go#L43-L43 to be high enough to make the test meaningful.

What if someone in future changes value there to e.g. 100ms?

I believe we should be explicit and setup here etcd with concrete high enough value to make this test proving that the feature we care about really works.

Reply:
Right